### ISSUE TITLE
Select distinct generated query doesn't use LIMIT

### ISSUE THREAD

**Issue by meotimdihia** [prisma/prisma#6259]
## Bug description

it seems Select Distinct generate Query doesn't use LIMIT.
https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#select-distinct  
I'm using the newest Prisma version.  
The problem is discussed here: https://github.com/prisma/prisma/discussions/6227

## How to reproduce

```
await prisma.md_chapters.findMany({
  distinct: ['md_comic_id'],
  select: {
    id: true,
    md_comic_id: true,
  },
  take: 100
});
```

## Environment & setup

- OS:  Windows 10
- Database: PostgreSQL
- Node.js version: ^v12
- Prisma version: Prisma ^2.19.0

**Comment by meotimdihia** [comment]
`take` and `skip` both don't work.

**Comment by pantharshit00** [comment]
I can reproduce this in `2.23.0-dev.28`

![image](https://user-images.githubusercontent.com/22195362/118406663-5ad41780-b69a-11eb-870d-60d47d165453.png)

**Comment by dpetrick** [comment]
This is not a bug, but by design. Prisma distinct queries are more expressive than what SQL generally offers - specifically, we offer `DISTINCT ON` semantics, which come from the Postgres world. For example, your query above can't be easily expressed in MySQL or SQL Server `DISTINCT`, because you're selecting the `id` field but don't include it in the `distinct`. 

Upcoming query optimization work will be able to fix some issues, e.g. if you're strictly following `DISTINCT` rules (selection set is identical to the distinct selection), we can fire efficient queries.

Tl;dr: We're aware, this will be handled somewhat more efficient in the future in some cases.

**Comment by janpio** [comment]
This is effectively a result of https://github.com/prisma/prisma/issues/14765.

**Comment by janpio** [comment]
If you are using PostgreSQL, you are in luck: A little while ago we added a new preview feature that transforms many of the `disctinct` queries to use actual `DISTINCT` like SQL queries under the hood, instead of doing the "distinct-ing" in memory. Read the instructions here: https://github.com/prisma/prisma/issues/14765#issuecomment-1845646423
